internal/command/deploy/deploy_first.go

```diff
@@ -9,12 +9,14 @@ import (
 	"github.com/superfly/flyctl/internal/prompt"
 )
 
-func (md *machineDeployment) provisionFirstDeploy(ctx context.Context) error {
+func (md *machineDeployment) provisionFirstDeploy(ctx context.Context, allocPublicIPs bool) error {
 	if !md.isFirstDeploy || md.restartOnly {
 		return nil
 	}
-	if err := md.provisionIpsOnFirstDeploy(ctx); err != nil {
-		fmt.Fprintf(md.io.ErrOut, "Failed to provision IP addresses, use `fly ips` commands to remmediate it. ERROR: %s", err)
+	if allocPublicIPs {
```

Comment:
this is fine but we already do checks inside `provisionIpsOnFirstDeploy`, better to have them all in one place as it is easy to follow for future ourselves.

Reply:
Done